marlowe-dashboard-client/src/Marlowe/PAB.purs

```diff
@@ -24,61 +24,93 @@ import Marlowe.Semantics (Contract, State, TransactionInput)
+to avoid confusion with *Marlowe* contracts:
+
+1. The `MarloweApp`. This is used to:
+   - "create" Marlowe contracts (i.e. to install them on the blockchain and distribute the role tokens)
+   - "apply-inputs" to a Marlowe contract (i.e. perform an action to move that contract forward)
+   - "redeem" payments made to a role token
+2. The `WalletCompanionApp`. Every wallet has one instance of this app installed in the PAB. It listens
+   continuously for payments of role tokens to that wallet, and updates its observable state with an
+   array of `(MarloweParams, MarloweData)`, one pair for each role token, hence one for each Marlowe
+   contract for which this wallet has a role.
+3. The `WalletFollowerApp`. Every wallet has zero or more istances of this app installed in the PAB. We
+   use this to "follow" a Marlowe contract. Once a contract has been "created", and we have its
+   `MarloweParams`, we use an instance of the `WalletFollowerApp` to track its history and status. Once
+   we have called the "follow" endpoint of this app (passing it the `MarloweParams`), its observable
+   state will tell us everything we need to know about the contract, and will be updated when it changes.
+-}
+data PlutusApp
+  = MarloweApp
+  | WalletCompanionApp
+  | WalletFollowerApp
+
+{-
+In order to activate instances of the Plutus "contracts" (or apps) in the PAB, we need to pass the path
+to the executable on the server. We get these paths from nix, which writes them to a `contracts.json` file.
+So we don't have to worry about this anywhere else, we provide here functions mapping `PlutusApp` to their
+paths (in the format that the PAB expects).
+-}
+foreign import marloweAppPath_ :: String
+
+foreign import walletCompanionAppPath_ :: String
+
+foreign import walletFollowerAppPath_ :: String
+
+plutusAppPath :: PlutusApp -> ContractExe
+plutusAppPath MarloweApp = ContractExe { contractPath: marloweAppPath_ }
+
+plutusAppPath WalletCompanionApp = ContractExe { contractPath: walletCompanionAppPath_ }
+
+plutusAppPath WalletFollowerApp = ContractExe { contractPath: walletFollowerAppPath_ }
+
+plutusAppType :: ContractExe -> Maybe PlutusApp
+plutusAppType exe
+  | exe == plutusAppPath MarloweApp = Just MarloweApp
+
+plutusAppType exe
+  | exe == plutusAppPath WalletCompanionApp = Just WalletCompanionApp
+
+plutusAppType exe
+  | exe == plutusAppPath WalletFollowerApp = Just WalletFollowerApp
+
+plutusAppType _ = Nothing
+
+-- in the PAB, transactions have a fixed cost of 10 lovelace; in the real node,transaction fees will vary,
+-- but this may still serve as a good approximation
 transactionFee :: BigInteger
 transactionFee = fromInt 10
 
--- FIXME: it appears that creating a contract currently requires three transactions,
--- but I believe it should be possible with one; check this with Alex
+-- FIXME: it appears that creating a contract currently requires three transactions, but I believe it
```

Comment:
Maybe we could replace this with an "effectful function" for getting the values from the BE, and for now to be hardcoded to this value